ddtrace/ext/aws.py

```diff
@@ -37,9 +37,13 @@ def add_span_arg_tags(
 ):
     # type: (...) -> None
     if endpoint_name not in EXCLUDED_ENDPOINT:
-        tags = dict((name, value) for (name, value) in zip(args_names, args) if name in args_traced)
-        flat_tags = flatten_dict(tags, exclude=EXCLUDED_ENDPOINT_TAGS.get(endpoint_name))
-        span.set_tags({k: truncate_arg_value(v) for k, v in flat_tags.items()})
+        exclude_set = EXCLUDED_ENDPOINT_TAGS.get(endpoint_name, frozenset())  # type: FrozenSet[str]
+        set_flattened_tags(
+            span,
+            items=((name, value) for (name, value) in zip(args_names, args) if name in args_traced),
```

Comment:
```suggestion
            items=(v for v in zip(args_names, args) if v[0] in args_traced),
```

maybe saves some time by not recreating tuples but probably negligible

Reply:
Counterintuitive Python 🙂 

![Screenshot 2021-06-30 at 09 47 22](https://user-images.githubusercontent.com/20231758/123931138-42893500-d988-11eb-8cc7-ba2552b7a00e.png)

Reply:
lol of course

Reply:
TBH this kind of micro-optimizations can blow up any time with future CPython runtime 😢